### PR TITLE
[10.x] Return exit status instead of booleans from `GeneratorCommand::handle()`

### PR DESCRIPTION
**Pull request by imdhemy**
This PR reopens #43915 and updates the commands that use the `handle()` method.

::: Note
A note should be added to the upgrade guide.

### CODE REVIEW

File: src/Illuminate/Console/GeneratorCommand.php [hunk 1]
```diff
@@ -258,12 +261,14 @@ protected function alreadyExists($rawName)
      *
      * @param  string  $name
      * @return string
+     *
+     * @throws BindingResolutionException
```
Comment (driesvints) [inline src/Illuminate/Console/GeneratorCommand.php:265]: Can you please remove all these `@throws` annotations you added everywhere? These methods don't throw this exception, they only cascade.

**Review by driesvints — CHANGES_REQUESTED** [review]

File: src/Illuminate/Console/GeneratorCommand.php (hunk 1 above)
Comment (imdhemy) [reply src/Illuminate/Console/GeneratorCommand.php:265]: Removed...

Comment (driesvints) [reply src/Illuminate/Console/GeneratorCommand.php:265]: @imdhemy there's still some in this PR (including this one).

File: tests/Console/GeneratorCommandTest.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+use Symfony\Component\Console\Exception\RuntimeException;
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Output\NullOutput;
+
+class GeneratorCommandTest extends TestCase
+{
+    /**
+     * @test
+     */
+    public function itShouldThrowIfNameIsNotPassedAsArgument(): void
```
Comment (driesvints) [inline tests/Console/GeneratorCommandTest.php:23]: Please use the same `test_` syntax as the rest of the test suite.

**Review by driesvints — CHANGES_REQUESTED** [review]

File: tests/Console/GeneratorCommandTest.php (hunk 2 above)
Comment (imdhemy) [reply tests/Console/GeneratorCommandTest.php]: @driesvints updated...

**Comment by taylorotwell** [comment]
This doesn't do anything to actually improve Laravel for end users, which is what I want to focus on.

**Pull request closed without merging**